dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataelement/DataElement.java

```diff
@@ -282,6 +285,35 @@ public boolean isFileType()
         return getValueType().isFile();
     }
 
+    /**
+     * Data type for use in analytics. Both text and date types are recognized
+     * as TEXT. Everything else is recognized as NUMERIC. Note that this needs
+     * to be based on the QueryModifiers.valueType if present.
+     */
+    public DataType getAnalyticsDataType()
```

Comment:
I think this is a bit unclear. I.e. in what analytics context do we think of everything which is non text or date as numeric?

Reply:
Hi @larshelge, thanks. When populating the analytics `value` and `textvalue` columns [here in JdbcAnalyticsTableManager](https://github.com/dhis2/dhis2-core/blob/master/dhis-2%2Fdhis-services%2Fdhis-service-analytics%2Fsrc%2Fmain%2Fjava%2Forg%2Fhisp%2Fdhis%2Fanalytics%2Ftable%2FJdbcAnalyticsTableManager.java#L252-L257), the text and date values go into `textvalue` and everything else goes into `value`.

I was patterning this code after the logic [here in QueryPlannerUtils](https://github.com/dhis2/dhis2-core/blob/master/dhis-2%2Fdhis-services%2Fdhis-service-analytics%2Fsrc%2Fmain%2Fjava%2Forg%2Fhisp%2Fdhis%2Fanalytics%2Fdata%2FQueryPlannerUtils.java#L104-L108) which uses `DataType.TEXT` to represent text and date values and `DataType.NUMERIC` for everything else. If there's a better way to code this or to comment it, I'm open to suggestions. It seemed a bit odd to be using a subset of `DataType` enum values for this special purpose, which is why I called this `getAnalyticsDataType` instead of just `getDataType` (as mentioned above under **DataElement**).

Other possibilities include:
1. add the analytics column name as a field in the DataType enum.
2. define a new 2-value enum, possibly added as a field in the DataType enum.
3. just use the column name in `DataElement` and `QueryPlannerUtils` instead of a two-valued `DataType`.
4. use a dummy column name for the value when the subexpression is generated, and have `JdbcAnalyticsManager` substitute the real column name (not as elegant for parsing, but keeps the analytics SQL knowledge more localized).

Most of these represent tradeoffs between (a) extending knowledge of analytics SQL columns into other parts of the code vs. (b) defining logical constructs in other parts of the code which are really there because analytics uses two columns for data values.

I'm open to suggestions.